diff -dur pysnmp-4.2.3.orig/pysnmp/smi/builder.py pysnmp-4.2.3/pysnmp/smi/builder.py
--- pysnmp-4.2.3.orig/pysnmp/smi/builder.py	2012-08-29 23:22:35.000000000 +0200
+++ pysnmp-4.2.3/pysnmp/smi/builder.py	2012-12-20 10:49:27.000000000 +0100
@@ -138,7 +138,7 @@
         
     def _getData(self, f, mode=None):
         data = self.__loader.get_data(os.path.join(self._srcName, f))
-        if sys.version_info[0] <= 2:
+        if sys.version_info[0] <= 2 and "b" not in mode:
             data = data.replace('\r\n', '\n')
         return data
 
@@ -160,7 +160,7 @@
         try:
             if f in os.listdir(self._srcName): # make FS case-sensitive
                 data = open(os.path.join(self._srcName, f), mode).read()
-                if sys.version_info[0] <= 2:
+                if sys.version_info[0] <= 2 and "b" not in mode:
                     data = data.replace('\r\n', '\n')
                 return data
         except OSError:
